lib/dataset-tools/requoting.js

```diff
@@ -193,7 +193,9 @@ function* getFunctions(program) {
 }
 
 function* getDevices(program) {
-    for (let fn of getFunctions(program)) {
+    let functions = getFunctions(program);
+    functions.next(); // discard the policy name (the first "function")
+    for (let fn of functions) {
```

Comment:
Actually this won't work - when we do single shot, we skip the $dialogue statement. You should check that the first token is "$dialogue" and skip the next token. And while you're there, you should fix getFunctions too...

Bonus points if you add unit tests for these.